aea/cli/common.py

```diff
@@ -496,3 +508,117 @@ def _find_item_locally(ctx, item_type, item_public_id) -> Path:
         sys.exit(1)
 
     return package_path
+
+
+def _compute_fingerprint(package_directory) -> Dict[str, str]:
+    hasher = IPFSHashOnly()
+    fingerprints = {}  # type: Dict[str, str]
+    # find all valid files of the package
+    all_files = [
+        x
+        for x in package_directory.glob("**/*")
+        if x.is_file() and not x.match("*__pycache__/*") and x.match("*.py")
+    ]
+
+    for file in all_files:
+        file_hash = hasher.get(str(file))
+        key = str(file.relative_to(package_directory))
+        assert key not in fingerprints  # nosec
+        fingerprints[key] = file_hash
+
+    return fingerprints
+
+
+def _check_package_fingerprints(
+    package_configuration: PackageConfiguration, package_directory: Path
+):
+    """Given a package configuration and a package directory, check whether
+    the hash of the files in the package directory match the one in the package configurations."""
+    expected_fingerprints = package_configuration.fingerprint
+    actual_fingerprints = _compute_fingerprint(package_directory)
+
+    if expected_fingerprints != actual_fingerprints:
```

Comment:
Can we have two types of warnings? I'd like it to say:
- in the case of non-matching fingerprints for vendorized projects: "Fingerprints for package {} do not match:\nExpected: {}\nActual:{}\nVendorized projects should not be tampered with, please revert any changes."
- in the case of non-matching fingerprints for root projects: "Fingerprints for package {} do not match:\nExpected: {}\nActual:{}\nPlease fingerprint the package in {} before continuing."

Reply:
yes, good idea!